syntax/syntax.go

```diff
@@ -25,6 +52,7 @@ func End(n Node) Position {
 
 // A File represents a Skylark file.
 type File struct {
+	Comments
```

Comment:
Please make each of these a pointer.  In the performance critical case (BUILD file loading), these Comment fields add 9 words (3 slices x 3 words) to every AST node, which more than doubles the size of the AST.  Making these fields pointers would add only 1 word to each node.  This does require the parser to make additional allocations when comments are requested, but applications should pay only for what they use (and reformatting tools typically load only a small handful of BUILD files in their lifetime, not tens of thousands).

Reply:
Of course, this assumes that Skylark comments will never become significant to the semantics, as they have (regrettably) in Go.

Reply:
Right, we don't intend to make comments significant.
(they might affect linters and other tools though, but not the interpreter)